from_json/as.ts

```diff
@@ -1,10 +1,10 @@
 // Copyright 2018-2020 Gamebridge.ai authors. All rights reserved. MIT license.
 
-import { FromJsonStrategy, Serializable } from "../serializable.ts";
+import { FromJsonStrategy, JsonObject, JsonValue, Serializable } from "../serializable.ts";
 
 /** revive data using `fromJson` on a subclass type */
 export function fromJsonAs<T>(
   type: T & { new (): Serializable },
 ): FromJsonStrategy {
-  return (value: T) => new type().fromJson(value);
+  return (value: JsonValue) => new type().fromJson(value as JsonObject);
```

Comment:
Having to cast `value as JsonObject` here makes me think maybe I'm doing something wrong, or having the separate distinction between JsonValue and JsonObject isn't as helpful as I'd hoped.

`fromJsonAs()` wants to convert a JsonObject to a user-defined type, but FromJsonStrategy can't be specific about JsonObject since it's also used for field-level deserialization. So the arrow function has to take a JsonValue, and yet we want to pass a nested JsonObject to fromJson().